os/drivers/usbhost/uvc/usbhost_uvc_driver.c

```diff
@@ -0,0 +1,2377 @@
+			INIT_QUEUE_HEAD(&intf->ep);
+
+			dq_addfirst(&intf->list, &priv->config.interface);
+
+			priv->config.ifcount++;
+		}
+		break;
+
+		case USB_DESC_TYPE_CS_INTERFACE: {
+			uvdbg("CS Interface descriptor\n");
+			if ((bIfClass == CC_VIDEO) && (bIfSubClass == USB_SUBCLASS_VIDEOCONTROL)) {
+				priv->intfnum = bIfNumber;
+				ret = uvc_parse_vc_descriptor(priv, (const unsigned char *)desc, desc->len);
+			} else if ((bIfClass == CC_VIDEO) && (bIfSubClass == USB_SUBCLASS_VIDEOSTREAMING)) {
+				ret = uvc_parse_vs_descriptor(priv, (const unsigned char *)desc, remaining, bIfNumber);
+			}
+			if (ret < 0) {
+				return -EINVAL;
+			}
+			/* TODO AUDIO CONTROL/STREAMING */
+		}
+		break;
+
+		case USB_DESC_TYPE_CSENDPOINT: {
+			uvdbg("CS Endpoint Interface descriptor\n");
+		}
+		break;
+
+		/* Endpoint descriptor.  Here, we expect one IN ISO endpoint and one IN INT endpoint */
+
+		case USB_DESC_TYPE_ENDPOINT: {
+			FAR struct usb_epdesc_s *epdesc = (FAR struct usb_epdesc_s *)configdesc;
+
+			uvdbg("Endpoint descriptor %d\n", epdesc->attr);
+			DEBUGASSERT(remaining >= USB_SIZEOF_EPDESC);
+
+			ep = kmm_zalloc(sizeof(*ep));
+			if (ep == NULL) {
+				udbg("Memory allocation failure!!\n");
+				return -ENOMEM;
+			}
+
+			memcpy(ep, epdesc, sizeof(struct usb_epdesc_s));
+
+			DEBUGASSERT(intf);
+
+			dq_addfirst(&ep->list, &intf->ep);
+		}
+		break;
+
+		/* Other descriptors are just ignored for now */
+
+		default:
+			uvdbg("Unknown descriptor %d\n", desc->type);
+			break;
+		}
+
+		/* Increment the address of the next descriptor */
+
+		configdesc += desc->len;
+		remaining -= desc->len;
+	}
+
+	return OK;
+}
+
+/****************************************************************************
+ * Name: uvc_scan_device
+ *
+ * Description:
+ *   The function is used to scan all the entities created on UVC descriptor
+ *   parsing and assign them to proper terminal(in/out).
+ *
+ *   This function is called from the usbhost_uvc_devinit() method.
+ *
+ * Input Parameters:
+ *   dev - A reference to the class instance.
+ *
+ * Returned Values:
+ *   On success, zero (OK) is returned. On a failure, a negated errno value is
+ *   returned indicating the nature of the failure
+ *
+ ****************************************************************************/
+int uvc_scan_device(FAR uvc_state_t *dev)
+{
+	struct uvc_video_chain *chain;
+	struct uvc_entity *term = queue_entry(dev->entities.head, struct uvc_entity, list);
+
+	/* Parse through the entity and add them to approprate terminal(in/out) */
+	for (; &term->list != (dq_entry_t *)&dev->entities; term = queue_entry(term->list.flink, struct uvc_entity, list)) {
+		if (!UVC_ENTITY_IS_OTERM(term)) {
+			continue;
+		}
+
+		if (term->chain.head || term->chain.tail) {
+			continue;
+		}
+
+		chain = kmm_zalloc(sizeof(*chain));
+		if (chain == NULL) {
+			udbg("Memory allocation failed!!\n");
+			return -ENOMEM;
+		}
+
+		INIT_QUEUE_HEAD(&chain->entities);
+		sem_init(&chain->ctrl_mutex, 0, 1);
```

Comment:
This semaphore is not being destroyed in "usbhost_uvc_destroy" function or elsewhere

Reply:
Thanks!! Added it